docs/sms-receive-how-to.md

```diff
@@ -0,0 +1,213 @@
+---
+title: Receive SMS
+topic: SMS
+type: how-to
+---
+
+# How to receive SMS
+
+## Overview
+
+To receive an SMS you will need to:
+* [Get credentials from Console](https://console.wgtwo.com/api-keys-redirect)
+* Write code that starts listening for incoming SMS
+* Ack each received SMS
+
+## Prerequisites
+
+### Token/credentials
+* You will need [credentials from Console](https://console.wgtwo.com/api-keys-redirect) to send SMS to/for users on your platform.
+
+### Install dependencies
+
+To add the dependencies, first you need to add the [Jitpack](https://jitpack.io) repository:
+```xml
+<repositories>
+    <repository>
+        <id>jitpack.io</id>
+        <url>https://jitpack.io</url>
+    </repository>
+</repositories>
+```
+
+Then you can add `messaging-grpc` and `common`:
+
+```xml
+<dependencies>
+    <dependency>
+        <groupId>com.github.working-group-two.wgtwoapis</groupId>
+        <artifactId>messaging-grpc</artifactId>
+        <version>91f3d656e6d890829e28f0ee7788359450325828</version>
+    </dependency>
+    <dependency>
+        <groupId>com.github.working-group-two.wgtwoapis</groupId>
+        <artifactId>common</artifactId>
+        <version>91f3d656e6d890829e28f0ee7788359450325828</version>
+    </dependency>
+</dependencies>
+```
+
+### Initialize your dependencies
+```kotlin
+import com.wgtwo.api.auth.Clients
+import com.wgtwo.api.common.OperatorToken
+
+val channel = Clients.createChannel(Clients.Environment.PROD)
+val credentials = OperatorToken("YOUR_CLIENT_ID", "YOUR_CLIENT_SECRET")
+val messageCoreStub = MessageCoreGrpc
+        .newStub(channel)
+        .withCallCredentials(credentials)
+        .withWaitForReady()
+```
+
+## Receive SMS
+```kotlin
+import com.wgtwo.example.Shared
+import io.grpc.Context
+import io.grpc.Status
+import io.grpc.StatusException
+import io.grpc.StatusRuntimeException
+import io.grpc.stub.StreamObserver
+import io.omnicate.messaging.protobuf.MessageCoreGrpc
+import io.omnicate.messaging.protobuf.Messagecore
+import org.slf4j.LoggerFactory
+import java.time.Duration
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeUnit
+
+interface SmsReceiver {
+    fun onReceived(sms: Messagecore.Message)
+}
+
+object ReceiveSmsService {
+    private val logger = LoggerFactory.getLogger(javaClass)
+    private val executor: ScheduledExecutorService = Executors.newSingleThreadScheduledExecutor()
+    private val context: Context.CancellableContext = Context.current().withCancellation()
+
+    private val channel = Clients.createChannel(Clients.Environment.PROD)
+    private val credentials = OperatorToken("YOUR_CLIENT_ID", "YOUR_CLIENT_SECRET")
+    private val messageCoreStub = MessageCoreGrpc
+        .newStub(Shared.channel)
+        .withCallCredentials(Shared.credentials)
+        .withWaitForReady()
+
+    /*
```

Comment:
it feels to melike this belongs outside of the code.
Maybe in another block after the code with the comment above it?

Reply:
Can do... I almost think it should come first 🤔. I'll try to see what makes sense and push another draft.